src/components/Layout/Topbar/TopBar.jsx

```diff
@@ -0,0 +1,53 @@
+import { HStack, Spacer, Icon, Heading, Flex } from '@chakra-ui/react';
+import PropTypes from 'prop-types'; // ES6
+import { FaTeamspeak } from 'react-icons/fa';
+
+import TopBarLink from './TopBarLink';
+
+import { useAuthContext } from 'context/AuthContext.jsx';
+
+function TopBar({ setIsLogged }) {
+  const { handleLogout, isUserLoggedIn } = useAuthContext();
+  const guestLinks = ['login', 'register'];
+  const userLinks = ['logout', 'my-profile'];
+  const links = isUserLoggedIn ? userLinks : guestLinks;
+
+  const navBarHandleLogout = () => {
+    handleLogout();
+    setIsLogged(false);
+  };
```

Comment:
No need to have a separate logout function for navBar, we can use the one from `AuthContext`, and if needed, we can move `setIsLogged(false)` to `AuthContext`'s `handleLogout` function

```suggestion

```